core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -118,17 +118,13 @@ public static void schedule(String threadName) {
   }
 
   /**
-   * Waits until the given thread can be executed.
+   * Waits for up to 10 seconds until the given thread can be executed.
    *
    * @param name a name of the thread to wait for
    * @throws InterruptedException if the waiting thread is interrupted
    */
   public static void await(String name) throws InterruptedException {
-    try (LockResource r = new LockResource(sLock)) {
-      while (!sTimers.containsKey(name)) {
-        sCondition.await();
-      }
-    }
+    await(name, 10, TimeUnit.SECONDS);
```

Comment:
Do I understand you correctly that you are acknowledging that you change introduces anti-symmetry that is weird?

To be honest, I don't see why our tests should be ever using await() method that can time out. If they do, their correctness relies on timing, which is something the scheduled heartbeat tries to prevent in the first place. If we want to prevent a test from hanging, it should have a method-level timeout instead of introducing ad-hoc time outs to scheduling primitives.